### PR TITLE
RegisterCommands method

### PR DESCRIPTION
**Pull request by tkow**
Close https://github.com/jmcdo29/nest-commander/issues/705.

I also add two sections named with Command Runner and Regsitere Commands to your docs and fix links broken I found in https://github.com/jmcdo29/nest-commander/pull/706/commits/4a3dd2d3e553539e62aeecf6d89cd5855e9f0d19.

# apps/docs/pages/commander.md
スクリーンショット 2022-12-25 14 57 27

# apps/docs/pages/api.md
スクリーンショット 2022-12-25 14 57 27

### CODE REVIEW

**Comment by tkow** [comment]
@jmcdo29
Please, close it if you don't like it as describing in https://github.com/jmcdo29/nest-commander/issues/705.

File: packages/nest-commander/src/command-runner.interface.ts [hunk 1]
```diff
@@ -23,7 +24,20 @@ export type InquirerKeysWithPossibleFunctionTypes =
 
 type InquirerQuestionWithoutFilter<T> = Omit<T, 'filter'>;
 
+type CommandRunnerClass = ClassProvider<CommandRunner> & typeof CommandRunner;
+
 export abstract class CommandRunner {
+  static registerCommands(meta: string = CommandMeta): CommandRunnerClass[] {
```
Comment (jmcdo29) [inline packages/nest-commander/src/command-runner.interface.ts:30]: Could we call this `registerWithSubCommands`? That way the intent is clear. The code is actually really clean, and if the command is clear enough then I don't see why we can't add it in. Thanks for adding the tests too, showing it works is really nice.

The only other thing I would ask is can you update the documentation? Just a new section in the [`Commander`](https://nest-commander.jaymcdoniel.dev/en/features/commander/) section would do

**Comment by jmcdo29** [comment]
Oh, also, would you mind running `pnpm changeset` and following the wizard to create a `minor` changeset for the `nest-commander` package? That will help automate the release of this once I merge it in.

File: apps/docs/src/pages/en/features/commander.md [hunk 2]
```diff
@@ -231,3 +231,25 @@ And now the `TaskRunner` is setup and ready to be used.
 The above command is meant to be a basic example, and should not be taken as a fully fleshed out CLI example. There's error handling, input validation, and security that should all be considered. Please do not use the above command in a production environment without adding the mentioned necessities **at the very least**.
 
 :::
+
+## Register Commands
+
+Though you'll find the details are in [factory page](./factory), you must register your all commands including sub commands as providers in your module class exists somewhere in Module tree registerd to `CommandFactory`. For convenience, given that we register exapmles in Sub Commands section and set them as providers in app.module.ts that set as root module to `CommandFactory`.
```
Comment (jmcdo29) [inline apps/docs/src/pages/en/features/commander.md:237]:
```suggestion
Though you'll find the implementation details in the [factory page](./factory), you must register all of your commands including the sub commands as providers in a module class that the `CommandFactory` ends up registering. For convenience, given that we register examples in Sub Commands section and set them as providers in `app.module.ts` that set as root module to `CommandFactory`.
```

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
A couple of grammatical revisions. Thanks for all the work!

**Comment by jmcdo29** [comment]
If everything (save for the sending coverage) passes, this PR will be great! Thanks again.

**Comment by tkow** [comment]
@jmcdo29 

Thank you for reviews, in fact, I'm not native English speaker, so I apologize for many grammatical mistakes.

I have no idea to fix CI / send-coverage, so could  I ask you to fix it?

BTW, I really think this library is awesome and I have been making many our command line tools standalone and want more known widely especially in my country, Japan. It can be used not for nest application but for command line tools builder.  If you don't mind it,  could you let me translate docs in ja locale?
And, sorry for that I make you work even thought today is Xmas. I'll give you  my gratitude sponsored page though it may be little for you. Have a nice day.

**Comment by jmcdo29** [comment]
> I have no idea to fix CI / send-coverage, so could I ask you how to fix it?

It's just because of a missing API token for Codacy. Don't worry about it :)

**Comment by jmcdo29** [comment]
> If you don't mind it, could you let me translate docs in ja locale when I have time?

Go for it! So long as you don't mind helping maintain them as well when changes happen :smile: I use [AstroDocs](https://github.com/withastro/astro/tree/latest/examples/docs?on=github#multiple-languages-support) for the documentation, which has multiple language support already built in.

**Comment by tkow** [comment]
@jmcdo29 

> It's just because of a missing API token for Codacy. Don't worry about it :)

So, Is there nothing I should change in this PR? 
> If everything (save for the sending coverage) passes, this PR will be great! Thanks again.
Sorry for I'm not familiar to nx e2e test, but I still have something to have to add extra configuration to do this.

**Comment by jmcdo29** [comment]
> So, Is there nothing I should change in this PR?

Nah, it's all good

**Review by jmcdo29 — APPROVED** [review]

**Pull request merged**